### PR TITLE
feat: Add new ptn modules `azd/insights-dashboard`.

### LINKED ISSUE #1206: [Module Proposal]: `avm/ptn/azd/insights-dashboard`
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Bicep

### Module Classification?

Pattern Module

### Module Name

avm/ptn/azd/insights-dashboard

### Module Details

Please add a Pattern Module named `avm/ptn/azd/insights-dashboard`, this module will create the following two resources:

1. `Application Insights instance` based on an existing Log Analytics workspace.
2. `Portal Dashboard` for an Application Insights instance.

For details, please refer to azd: 
https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/monitor/applicationinsights.bicep

https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/monitor/applicationinsights-dashboard.bicep

CC: @jongio

### Do you want to be the owner of this module?

No

### Module Owner's GitHub Username (handle)

jongio

### (Optional) Secondary Module Owner's GitHub Username (handle)

_No response_

### PR DESCRIPTION
**Pull request by Menghua1**
## Description
Fixes https://github.com/Azure/Azure-Verified-Modules/issues/1206.

## Pipeline Reference

| Pipeline |
| -------- |
|  [avm.ptn.azd.insights-dashboard](https://github.com/Menghua1/bicep-registry-modules/actions/workflows/avm.ptn.azd.insights-dashboard.yml)     |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

@jongio for notification.

### CODE REVIEW

File: avm/ptn/azd/insights-dashboard/tests/e2e/defaults/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      dashboardName: '${uniqueString(deployment().name, resourceLocation)}-test-dashboard-${serviceShort}'
```
Comment (AlexanderSehr) [inline avm/ptn/azd/insights-dashboard/tests/e2e/defaults/main.test.bicep:53]: This in an optional value and hence should not be part of the `defaults` test. 
I'd propably recommend to just duplicate this test, have one with the dashboard (and call it e.g., `max` - see other modules), and have this one still as the `defaults` test just with not the dashboard name.

File: avm/ptn/azd/insights-dashboard/tests/e2e/defaults/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-log-analytics-${serviceShort}'
+  }
+}
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
```
Comment (AlexanderSehr) [inline avm/ptn/azd/insights-dashboard/tests/e2e/defaults/main.test.bicep:40]:
```suggestion
resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
  name: resourceGroupName
  location: resourceLocation
}

module dependencies 'dependencies.bicep' = {
  name: '${uniqueString(deployment().name, resourceLocation)}-test-dependencies'
  scope: resourceGroup
  params: {
    name: '${uniqueString(deployment().name, resourceLocation)}-test-log-analytics-${serviceShort}'
  }
}
```

Just aligning it to the order of deployment

File: avm/ptn/azd/insights-dashboard/main.bicep [hunk 3]
```diff
@@ -0,0 +1,109 @@
+metadata name = 'Application Insights Components'
```
Comment (AlexanderSehr) [inline avm/ptn/azd/insights-dashboard/main.bicep:1]: Softblocker until MAR file is updated
cc: @matebarabas

Comment (matebarabas) [reply avm/ptn/azd/insights-dashboard/main.bicep:1]: FYI, the required record is currently in PR, along with all the records of the other AZD ptn modules. I'll let you know when it's merged.

**Comment by Menghua1** [comment]
@AlexanderSehr  I have updated based on your comments, please re-review.

File: avm/ptn/azd/insights-dashboard/tests/e2e/defaults/main.test.bicep (hunk 1 above)
Comment (Menghua1) [reply avm/ptn/azd/insights-dashboard/tests/e2e/defaults/main.test.bicep]: Removed optional value `dashboardName` and added test for max.

**Review by AlexanderSehr — COMMENTED** [review]
Hey @Menghua1,
nothing to add from my side. Great work. Now we just need to wait for the PR to MAR to be merged.

PS: If I'd want to be nitpicky you could remove the `enableTelemetry` parameter from the max test as `true` is anyways the default & regenerate the readme. But ... it's not mandatory 😄

**Comment by Menghua1** [comment]
> PS: If I'd want to be nitpicky you could remove the `enableTelemetry` parameter from the max test as `true` is anyways the default & regenerate the readme. But ... it's not mandatory 😄

Hey, @AlexanderSehr,
Thank you very much. The `enableTelemetry` parameter has been removed from the maximum test.😄😄😄

In addition, I would like to ask if this PR can be merged as soon as possible. The other ptn modules we are working on depend on this module. If this is not merged, the progress of our other ptn modules will be blocked.😄😄😄

**Comment by AlexanderSehr** [comment]
> > PS: If I'd want to be nitpicky you could remove the `enableTelemetry` parameter from the max test as `true` is anyways the default & regenerate the readme. But ... it's not mandatory 😄
> 
> Hey, @AlexanderSehr, Thank you very much. The `enableTelemetry` parameter has been removed from the max test.😄😄😄
> 
> In addition, I would like to ask if this PR can be merged as soon as possible. The other ptn modules we are working on depend on this module. If this is not merged, the progress of our other ptn modules will be blocked.😄😄😄

100% we're only waiting for the team owning the MAR file to review & merge @matebarabas's PR. Once it's in, we can go right ahead.

**Comment by matebarabas** [comment]
@AlexanderSehr, @Menghua1, good news: the MAR manifest has been updated with the required record(s) for this module. You can take the next steps accordingly. Thanks.

**Comment by matebarabas** [comment]
@Menghua1, please make sure the following disclaimer is added to the `readme.md` of this module - preferably somewhere close to the top: "*This module is not intended for broad, generic use, as it was designed to cater for the requirements of the AZD CLI product. Feature requests and bug fix requests are welcome if they support the development of the AZD CLI but may not be incorporated if they aim to make this module more generic than what it needs to be for its primary use case.*"

**Comment by AlexanderSehr** [comment]
> @Menghua1, please make sure the following disclaimer is added to the `readme.md` of this module - preferably somewhere close to the top: "_This module is not intended for broad, generic use, as it was designed to cater for the requirements of the AZD CLI product. Feature requests and bug fix requests are welcome if they support the development of the AZD CLI but may not be incorporated if they aim to make this module more generic than what it needs to be for its primary use case._"

@Menghua1,
to do so, please do it like I did for the `container-apps` module you can see [here](https://github.com/Azure/bicep-registry-modules/pull/3221/files), i.e., add the disclaimer to the module's description in the `main.bicep` and then just re-generate the files using the `Set-AVMModule` script

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**